api/v1/replication-contraints.go

```diff
@@ -0,0 +1,120 @@
+}
+
+// handleSetReplicationConstraints godoc
+//
+//	@Summary		Creates or updates ReplicationConstraints associated with a tenant and collection
+//	@Param			token header string true "Auth token"
+//	@Param			ReplicationConstraints body
+//	@Produce		json
+//	@Success		200	{object}	ResponseEnvelope{response=ReplicationConstraints}
+//	@Router			/collections/:collectionUUID/replication-constraints [put]
+func (a *apiV1) handleSetReplicationConstraints(c echo.Context) error {
+	var replicationConstraints []ReplicationConstraintResponse
+
+	err := json.NewDecoder(c.Request().Body).Decode(&replicationConstraints)
+
+	if err != nil {
+		return c.JSON(http.StatusInternalServerError, CreateErrorResponseEnvelope(c, http.StatusInternalServerError, err.Error()))
+	}
+
+	err = a.db.Transaction(func(tx *gorm.DB) error {
+		for _, replicationConstraint := range replicationConstraints {
+			constraint := &db.ReplicationConstraint{CollectionID: uuid.MustParse(c.Param("collectionUUID")), ConstraintID: replicationConstraint.ConstraintID, ConstraintMax: replicationConstraint.ConstraintMax}
+
+			res := tx.Save(&constraint)
+
+			if res.Error != nil {
+				return err
+			}
+		}
+
+		return nil
+	})
+
+	if err != nil {
+		return c.JSON(http.StatusInternalServerError, CreateErrorResponseEnvelope(c, http.StatusInternalServerError, err.Error()))
+	}
+
+	return c.JSON(http.StatusOK, CreateSuccessResponseEnvelope(c, "Successfully created or updated replication constraint"))
+}
+
+// handleDeleteReplicationConstraints godoc
+//
+//	@Summary		Delete ReplicationConstraints used by a tenant
+//	@Param			token header string true "Auth token"
+//	@Produce		json
+//	@Success		200	{object}	ResponseEnvelope{response=ReplicationConstraints}
+//	@Router			/collections/:collectionUUID/replication-constraints/:replicationConstraintId [delete]
+func (a *apiV1) handleDeleteReplicationConstraints(c echo.Context) error {
+	constrainId := c.Param("replicationConstraintId")
+	id, err := strconv.Atoi(constrainId)
+
+	if err != nil {
+		return c.JSON(http.StatusInternalServerError, CreateErrorResponseEnvelope(c, http.StatusInternalServerError, err.Error()))
+	}
+
+	err = a.db.Delete(&ReplicationConstraint{CollectionID: uuid.MustParse(c.Param("collectionUUID")), ConstraintID: db.ID(id)}).Error
+
+	if err != nil {
+		return c.JSON(http.StatusInternalServerError, CreateErrorResponseEnvelope(c, http.StatusInternalServerError, err.Error()))
+	}
+
+	return c.JSON(http.StatusOK, CreateSuccessResponseEnvelope(c, "Successfully deleted replication constraint"))
+}
+
+// handleGetReplicationConstraints godoc
+//
+//	@Summary		Get ReplicationConstraints used by a tenant
+//	@Param			token header string true "Auth token"
+//	@Produce		json
+//	@Success		200	{object}	ResponseEnvelope{response=ReplicationConstraints}
+//	@Router			/collections/:collectionUUID/replication-constraints/:replicationConstraintId [get]
+func (a *apiV1) handleGetReplicationConstraints(c echo.Context) error {
```

Comment:
This should return all the replication constraints for the Tenant instead of being a `GetByID` type of API. i.e, return []ReplicationConstraintResponse with all of them for the tenant

Reply:
Should we add tenant id into the table or do the joins?

Reply:
This is totally my bad - we still want to scope this to return the `constraints` for a specific `collection`, not all for the tenant across all collections.

So you can still capture the value `collectionUUID` from the request, and use that in `db.ReplicationConstraint.Find()`, matching  on `CollectionUUID`. shouldn't need to have the TenantID on the ReplicationConstraints table as there's already a mapping with them to Collections.

Sorry about that!